bundles/org.openhab.binding.dwdpollenflug/src/main/java/org/openhab/binding/dwdpollenflug/internal/handler/DWDPollenflugBridgeHandler.java

```diff
@@ -0,0 +1,166 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dwdpollenflug.internal.handler;
+
+import static org.openhab.binding.dwdpollenflug.internal.DWDPollenflugBindingConstants.*;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.dwdpollenflug.internal.DWDPollingException;
+import org.openhab.binding.dwdpollenflug.internal.config.DWDPollenflugBridgeConfiguration;
+import org.openhab.binding.dwdpollenflug.internal.dto.DWDPollenflug;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DWDPollenflugBridgeHandler} is the handler for bridge thing
+ *
+ * @author Johannes DerOetzi Ott - Initial contribution
+ */
+@NonNullByDefault
+public class DWDPollenflugBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(DWDPollenflugBridgeHandler.class);
+
+    private @NonNullByDefault({}) DWDPollenflugBridgeConfiguration bridgeConfig = null;
+
+    private final DWDPollenflugPolling pollingJobRunnable;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable DWDPollenflug pollenflug;
+
+    private boolean ignoreConfigurationUpdate;
+
+    private final List<DWDPollenflugRegionListener> regionListeners = new CopyOnWriteArrayList<>();
+
+    public DWDPollenflugBridgeHandler(Bridge bridge, HttpClient client) {
+        super(bridge);
+        pollingJobRunnable = new DWDPollenflugPolling(this, client);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing DWD Pollenflug bridge handler");
+        bridgeConfig = getConfigAs(DWDPollenflugBridgeConfiguration.class);
+
+        if (bridgeConfig.isValid()) {
+            onUpdate();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    private synchronized void onUpdate() {
+        if (regionListeners.isEmpty()) {
+            stopPolling();
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            startPolling();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        stopPolling();
+    }
+
+    public void startPolling() {
+        if (pollingJob == null || pollingJob.isCancelled()) {

```

Comment:
What is the advantage of this?

Reply:
The advantage is more below, where you access the field a second time after the null check. There is no guarantee that the field is still non-null (the compiler should show a warning). This can be avoided by local variables as nothing outside can change it.

Reply:
Please check my fix!